go/libraries/doltcore/sqle/index/index_lookup.go

```diff
@@ -205,59 +220,214 @@ func NewLookupBuilder(part sql.Partition, idx DoltIndex, projections []uint64, p
 		return &keylessLookupBuilder{
 			baseLookupBuilder: base,
 		}
-	case idx.coversColumns(p.durableState, projections) || idx.ID() == "PRIMARY":
-		return &coveringLookupBuilder{
-			baseLookupBuilder: base,
-		}
+	case idx.coversColumns(p.durableState, projections):
+		return newCoveringLookupBuilder(base)
 	default:
-		return base
+		return newNonCoveringLookupBuilder(p, base)
+	}
+}
+
+func newCoveringLookupBuilder(b *baseLookupBuilder) *coveringLookupBuilder {
+	var keyMap, valMap, ordMap val.OrdinalMapping
+	if b.idx.IsPrimaryKey() {
+		keyMap, valMap, ordMap = primaryIndexMapping(b.idx, b.sch, b.projections)
+	} else {
+		keyMap, ordMap = coveringIndexMapping(b.idx, b.projections)
+	}
+	return &coveringLookupBuilder{
+		baseLookupBuilder: b,
+		keyMap:            keyMap,
+		valMap:            valMap,
+		ordMap:            ordMap,
 	}
 }
 
+func newNonCoveringLookupBuilder(p rangePartition, b *baseLookupBuilder) *nonCoveringLookupBuilder {
+	primary := durable.ProllyMapFromIndex(p.durableState.Primary)
+	priKd, _ := primary.Descriptors()
+	tbBld := val.NewTupleBuilder(priKd)
+	pkMap := ordinalMappingFromIndex(b.idx)
+	keyProj, valProj, ordProj := projectionMappings(b.idx.Schema(), b.projections)
+	return &nonCoveringLookupBuilder{
+		baseLookupBuilder: b,
+		pri:               primary,
+		priKd:             priKd,
+		pkBld:             tbBld,
+		pkMap:             pkMap,
+		keyMap:            keyProj,
+		valMap:            valProj,
+		ordMap:            ordProj,
+	}
+}
+
+var _ LookupBuilder = (*baseLookupBuilder)(nil)
+var _ LookupBuilder = (*nomsLookupBuilder)(nil)
+var _ LookupBuilder = (*coveringLookupBuilder)(nil)
+var _ LookupBuilder = (*keylessLookupBuilder)(nil)
+var _ LookupBuilder = (*nonCoveringLookupBuilder)(nil)
+
+// baseLookupBuilder is a common lookup builder for prolly covering and
```

Comment:
This is a little mind-bending where this `baseLookupBuilder` implements this interface for the prolly case, but it also still gets embedded in a nomsLookupBuilder. I think it would be more clear at this point if the common fields went on something else and this thing should be a prollyLookupBuilder or something.

Reply:
`baseLookupBuilder.RowIter` is never called so I will panic there. I used `nonCoveringLookupBuilder` for `prollyIndexLookup` because it confused me less.